nifi-nar-bundles/nifi-soap-bundle/nifi-soap-processors/src/main/java/org/apache/nifi/processors/soap/GetSOAP.java

```diff
@@ -214,106 +276,178 @@ public void onScheduled(final ProcessContext context) {
+        if (authByHeader != null && !"".equals(authByHeader)) {
+            Map<String, Object> map = null;
+            try {
+                map = mapper.readValue(authByHeader,
+                                       new TypeReference<Map<String, Object>>() {
+                                       });
+            }
+            catch (IOException e) {
+                getLogger().error(e.getMessage(), e);
+            }
 
+            authHeader = createAuthHeader(null, map);
+        } else if (null != userName && null != password && !userName.isEmpty() && !password.isEmpty()) {
             HttpTransportPropertiesImpl.Authenticator
                     auth = new HttpTransportPropertiesImpl.Authenticator();
             auth.setUsername(userName);
             auth.setPassword(password);
             options.setProperty(org.apache.axis2.transport.http.HTTPConstants.AUTHENTICATE, auth);
         }
+
         try {
             serviceClient = new ServiceClient();
             serviceClient.setOptions(options);
-        } catch (AxisFault axisFault) {
-            getLogger().error("Failed to create webservice client, please check that the service endpoint is available and " +
-                    "the property is valid.", axisFault);
+        }
+        catch (AxisFault axisFault) {
+            getLogger().error(
+                    "Failed to create webservice client, please check that the service endpoint is available and " +
+                    "the property is valid.",
+                    axisFault);
             throw new ProcessException(axisFault);
         }
     }
 
+    protected OMElement createAuthHeader(OMElement parent, Map<String, Object> map) {
+        OMElement element = null;
+        if (map != null) {
+            for (String key : map.keySet()) {
+                element = fac.createOMElement(key, omNamespace);
+                Object value = map.get(key);
+                if (value instanceof String) {
+                    element.setText(value.toString());
+                    if (parent != null) {
+                        parent.addChild(element);
+                    }
+                } else {
+                    OMElement childElement = createAuthHeader(element, (Map) value);
+                    element.addChild(childElement);
+                }
+            }
+        }
+        return element;
+    }
+
     @OnStopped
     public void onStopped(final ProcessContext context) {
         try {
             serviceClient.cleanup();
-        } catch (AxisFault axisFault) {
+        }
+        catch (AxisFault axisFault) {
             getLogger().error("Failed to clean up the web service client.", axisFault);
             throw new ProcessException(axisFault);
         }
     }
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
```

Comment:
well, i had it defined to be input required. if we are changing to input allowed, then yes.